Feature/practice 3

ken-tsukamoto opened the pull request:
・constantファイル作成
・Creatureファイル作成

ken-tsukamoto committed:
・GameManagerからViewに分割
・constantファイル作成
・Creatureファイル作成

tashima-t commented on Unity/Assets/Scripts/ChangeFieldImageView.cs:
```diff
@@ -0,0 +1,68 @@
+    {
+        if (_fieldText.text == FieldConstant.Tokyo)
+        {
+            _fieldText.text = FieldConstant.Savannah;
+            _fieldImage.sprite = _fieldImages[(int)FieldConstant.Field.Savannah];
```
この処理をまず、関数にまとめられそうです。
_fieldText.text =
_fieldImage.sprite =

ken-tsukamoto replied:
この処理からまず修正をします。

tashima-t commented on Unity/Assets/Scripts/ChangeFieldImageView.cs:
```diff
@@ -0,0 +1,68 @@
+
+    public void ChangeNextFieldImage()
+    {
+        if (_fieldText.text == FieldConstant.Tokyo)
```
ChangeFieldImageView
if分の部分をListで管理したほうが見やすく、少ない文で済みそうです。

tashima-t commented on Unity/Assets/Scripts/ChangeFieldImageView.cs:
```diff
@@ -0,0 +1,68 @@
+
+    [SerializeField] Text _fieldText;
+
+    public void ChangeFieldImage(){
```
関数名と実装内容は一致させた方がよいです。

ken-tsukamoto replied:
上記3つのコメントを全て修正しました。

ken-tsukamoto committed:
＜ChangeFieldImageViewの改善＞
・ _fieldText.text = と _fieldImage.sprite = を関数にまとめる
・ if文の部分をListで管理
・ 関数名と実装内容を一致

tashima-t commented on Unity/Assets/Scripts/ChangeFieldImageView.cs:
```diff
@@ -13,56 +13,44 @@ public class ChangeFieldImageView : MonoBehaviour
-            _fieldText.text = FieldConstant.Space;
-            _fieldImage.sprite = _fieldImages[(int)FieldConstant.Field.Space];
+            _fieldNumber++;
+            if (_fieldNumber == 4)
```
マジックナンバーを無くしましょう

tashima-t commented on Unity/Assets/Scripts/ChangeFieldImageView.cs:
```diff
@@ -13,56 +13,44 @@ public class ChangeFieldImageView : MonoBehaviour
-        }
-        else if (_fieldText.text == FieldConstant.PacificOcean)
+    void ChangeFieldImage(bool next){
+        List<string> _changeFields = new List<string>()
```
ローカルに書いておきましょう。
名前にChangeは関係ないので消しましょう。
ここに書いてしまうと呼ばれる度にnewしてしまいます。

ken-tsukamoto committed:
・ マジックナンバーの消去
・ Listの変更

takahiro-imai-samurai commented on Unity/Assets/Scripts/ChangeFieldImageView.cs:
```diff
@@ -0,0 +1,57 @@
+    {
+        _fieldText.text = fieldName;
+        _fieldImage.sprite = _fieldImages[fieldNumber];
+    }
```
引数がなくても問題なさそうに見えますね
```suggestion
    void SetTextAndImage(int fieldNumber)
    {
        _fieldText.text = _fieldsName[fieldNumber];
        _fieldImage.sprite = _fieldImages[fieldNumber];
    }
```

takahiro-imai-samurai commented on Unity/Assets/Scripts/BattleView.cs:
```diff
@@ -0,0 +1,224 @@
+            _opponentScaledPower = opponent.power;
+        }
+
+        _playerCreatureText.text = player.field + "\n" + Power + ":" +
+            _playerScaledPower + "\n" + player.name;
```
文字の連結は、+演算子だとメモリの確保が行われるため以下の方法をおすすめします。
```suggestion
        _playerCreatureText.text = $"{player.field}\n"{Power}:{_playerScaledPower}\n{player.name}";
```

参考: https://www.sejuku.net/blog/50133

ken-tsukamoto committed:
＜BattleView, ChangeFieldImageViewの修正＞
・ SetTextAndImageの引数を消去 (CFIV)
・ 文字列の連結に文字列補完を使用 (BV)
・ JudgeBattleを少し変更 (BV)

ken-tsukamoto committed:
・ SetCreatureを修正 (BattleView)
・ 命名を修正

ken-tsukamoto committed:
・ SetCreatureをInitializeCreatureに変更
